client/src/document/index.tsx

```diff
@@ -41,6 +40,9 @@ export function Document(props /* TODO: define a TS interface for this */) {
     async (url) => {
       const response = await fetch(url);
       if (!response.ok) {
+        if (response.status === 404) {
+          throw new Error(`${response.status} on ${url}: Page not found`);
```

Comment:
Unlike the error 2 lines down on line 47, the `response.text()` is the WHOLE HTML body so you don't want to include that in the error message. ...when it's a 404.